part of upstream commit e965d5147e216ceb4157ad478cd271af96866e77 as it seems to
be multiple fixes in one ...

https://bugs.gentoo.org/269307

diff --git a/nptl/init.c b/nptl/init.c
index d0f1fc3..ba3caee 100644
--- a/nptl/init.c
+++ b/nptl/init.c
@@ -260,7 +260,7 @@ extern void **__libc_dl_error_tsd (void) __attribute__ ((const));
 
 
 /* This can be set by the debugger before initialization is complete.  */
-static bool __nptl_initial_report_events;
+static bool __nptl_initial_report_events __attribute_used__;
 
 void
 __pthread_initialize_minimal_internal (void)
diff --git a/nptl/pthread_create.c b/nptl/pthread_create.c
index 5fcc72c..c693979 100644
--- a/nptl/pthread_create.c
+++ b/nptl/pthread_create.c
@@ -1,4 +1,4 @@
-/* Copyright (C) 2002-2007, 2008 Free Software Foundation, Inc.
+/* Copyright (C) 2002-2007,2008,2009 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Ulrich Drepper <drepper@redhat.com>, 2002.
 
@@ -40,10 +40,10 @@ static int start_thread (void *arg);
 int __pthread_debug;
 
 /* Globally enabled events.  */
-static td_thr_events_t __nptl_threads_events;
+static td_thr_events_t __nptl_threads_events __attribute_used__;
 
 /* Pointer to descriptor with the last event.  */
-static struct pthread *__nptl_last_event;
+static struct pthread *__nptl_last_event __attribute_used__;
 
 /* Number of threads running.  */
 unsigned int __nptl_nthreads = 1;
